benchmarks/cudf-merge.py

```diff
@@ -0,0 +1,325 @@
+        df = cudf.DataFrame(
+            {
+                "key": cupy.arange(start, stop=stop, dtype="int64"),
+                "payload": cupy.arange(local_size, dtype="int64"),
+            }
+        )
+    else:
+        # Other dataframe
+        #
+        # "key" column matches values from the build dataframe
+        # for a fraction (`frac_match`) of the entries. The matching
+        # entries are perfectly balanced across each partition of the
+        # "base" dataframe.
+        #
+        # "payload" column is a random permutation of the chunk_size
+
+        # Step 1. Choose values that DO match
+        sub_local_size = local_size // num_chunks
+        sub_local_size_use = max(int(sub_local_size * frac_match), 1)
+        arrays = []
+        for i in range(num_chunks):
+            bgn = (local_size * i) + (sub_local_size * i_chunk)
+            end = bgn + sub_local_size
+            ar = cupy.arange(bgn, stop=end, dtype="int64")
+            arrays.append(cupy.random.permutation(ar)[:sub_local_size_use])
+        key_array_match = cupy.concatenate(tuple(arrays), axis=0)
+
+        # Step 2. Add values that DON'T match
+        missing_size = local_size - key_array_match.shape[0]
+        start = local_size * num_chunks + local_size * i_chunk
+        stop = start + missing_size
+        key_array_no_match = cupy.arange(start, stop=stop, dtype="int64")
+
+        # Step 3. Combine and create the final dataframe chunk
+        key_array_combine = cupy.concatenate(
+            (key_array_match, key_array_no_match), axis=0
+        )
+        df = cudf.DataFrame(
+            {
+                "key": cupy.random.permutation(key_array_combine),
+                "payload": cupy.arange(local_size, dtype="int64"),
+            }
+        )
+    return df
+
+
+async def worker(rank, eps, args):
+    # Setting current device and make RMM use it
+    dev_id = args.devs[rank % len(args.devs)]
+    cupy.cuda.runtime.setDevice(dev_id)
+    rmm.reinitialize(
+        pool_allocator=True, devices=dev_id, initial_pool_size=args.rmm_init_pool_size
+    )
+
+    # Make cupy use RMM
+    cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)
+
+    df1 = generate_chunk(rank, args.chunk_size, args.n_chunks, "build", args.frac_match)
+    df2 = generate_chunk(rank, args.chunk_size, args.n_chunks, "other", args.frac_match)
+
+    # Let's warmup and sync before benchmarking
+    await distributed_join(args, rank, eps, df1, df2)
+    await barrier(rank, eps)
+
+    if args.cuda_profile:
+        cupy.cuda.profiler.start()
+
+    if args.profile:
+        pr = cProfile.Profile()
+        pr.enable()
+
+    timings = []
+    t1 = clock()
+    df3 = await distributed_join(args, rank, eps, df1, df2, timings)
+    await barrier(rank, eps)
+    took = clock() - t1
+
+    if args.profile:
+        pr.disable()
+        s = io.StringIO()
+        ps = pstats.Stats(pr, stream=s)
+        ps.dump_stats("%s.%0d" % (args.profile, rank))
```

Comment:
I think this outputs to a binary format.  Are you running through snakeviz as a post processing task ?

Reply:
Yes, snakeviz or something like `python -m pstats report.prof`

Reply:
cool